examples/platform/nrfconnect/app/Service.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *   This file implements the service publishing code for example usage.
+ */
+
+#include "Service.h"
+
+#include <string.h>
+
+#include <platform/CHIPDeviceLayer.h>
+#include <logging/log.h>
+#include <zephyr.h>
+
+#if CHIP_ENABLE_OPENTHREAD
+#include <openthread/message.h>
+#include <openthread/udp.h>
+#include <platform/OpenThread/OpenThreadUtils.h>
+#include <platform/ThreadStackManager.h>
+#include <platform/nrfconnect/ThreadStackManagerImpl.h>
+#endif
+
+LOG_MODULE_DECLARE(serv);
+
+// Transport Callbacks
+namespace {
+
+char deviceName[128];
+constexpr uint16_t kUDPBroadcastPort = 23367;
+
+} // namespace
+
+void SetDeviceName(const char * newDeviceName)
```

Comment:
This doesn't seem to be used. How about using ConnectivityMgr().GetBLEDeviceName()? deviceName array could be of size CONFIG_BT_DEVICE_NAME_MAX then.

Also, we might actually set BLEDeviceName() to "CHIP Lock"/"CHIP Lighting" or so (though we may do that in a separate commit.

Reply:
For now I removed this method. I will set device names in the separate PR.